### PR TITLE
Adjust precision of Microsoft.VisualBasic.Tests.FinancialTests based on platform

### LINKED ISSUE #39380: Test failure: Microsoft.VisualBasic.Tests.FinancialTests/DDB(Cost: 10000, Salvage: 4350, Life: 84, Period: 35, Factor: 2, expectedOld: 57.3268063538875, expected)
Test `Microsoft.VisualBasic.Tests.FinancialTests/DDB(Cost: 10000, Salvage: 4350, Life: 84, Period: 35, Factor: 2, expectedOld: 57.3268063538875, expected)` has failed.

Message :
```
Assert.Equal() Failure
Expected: 57.32680635388748
Actual:   57.32680635388746
```
Stack Trace :
```
   at Microsoft.VisualBasic.Tests.FinancialTests.AreEqual[T](T expectedOld, T expectedNew, T actual) in /_/src/Microsoft.VisualBasic.Core/tests/FinancialTests.cs:line 20
   at Microsoft.VisualBasic.Tests.FinancialTests.DDB(Double Cost, Double Salvage, Double Life, Double Period, Double Factor, Double expectedOld, Double expectedNew) in /_/src/Microsoft.VisualBasic.Core/tests/FinancialTests.cs:line 38
```
Build: -[20190710.51](https://mc.dot.net/#/user/dotnet-bot/pr~2Fdotnet~2Fcorefx~2Frefs~2Fheads~2Fmaster/test~2Ffunctional~2Fcli~2F~2F/20190710.51)(master)

Failing configuration:

- Windows.10.Arm64.Open-arm64-Release

Details:
https://mc.dot.net/#/user/dotnet-bot/pr~2Fdotnet~2Fcorefx~2Frefs~2Fheads~2Fmaster/test~2Ffunctional~2Fcli~2F~2F/20190710.51/workItem/Microsoft.VisualBasic.Core.Tests/analysis/xunit/Microsoft.VisualBasic.Tests.FinancialTests~2FDDB(Cost:%2010000,%20Salvage:%204350,%20Life:%2084,%20Period:%2035,%20Factor:%202,%20expectedOld:%2057.3268063538875,%20expected)

### LINKED ISSUE #39393: Allow for some variance in Microsoft.VisualBasic.Tests.FinancialTests
See https://github.com/dotnet/corefx/pull/39391/files#r302621690

### PR DESCRIPTION
**Pull request by cston**
Fixes #39380 
Fixes #39393

### CODE REVIEW

File: src/Microsoft.VisualBasic.Core/tests/FinancialTests.cs [hunk 1]
```diff
@@ -9,35 +9,33 @@ namespace Microsoft.VisualBasic.Tests
...
-        /// The accuracy of some numeric parsing and formatting has been increased,
-        /// so this will use correct value if the tests run on Full Framework.
+        /// The accuracy of some numeric parsing and formatting depends on the platform.
         /// </summary>
-        private static void AreEqual<T>(T expectedOld, T expectedNew, T actual)
+        private static void AreEqual(double expected, double actual)
         {
-            T expected = PlatformDetection.IsFullFramework ? expectedOld : expectedNew;
-            Assert.Equal(expected, actual);
+            Assert.Equal(expected, actual, precision);
```
Comment (danmosemsft) [inline src/Microsoft.VisualBasic.Core/tests/FinancialTests.cs]: I'd be tempted to just inline this, I don't think the separate method is helpful, it just hides what's going on. Your call obviously

File: src/Microsoft.VisualBasic.Core/tests/FinancialTests.cs [hunk 2]
```diff
@@ -9,35 +9,26 @@ namespace Microsoft.VisualBasic.Tests
 {
     public class FinancialTests
     {
-        private static bool IsNotArmOrAlpine() => !PlatformDetection.IsArmOrArm64Process && !PlatformDetection.IsAlpine;
+        // The accuracy of some numeric parsing and formatting depends on the platform.
+        private static readonly int precision = PlatformDetection.IsFullFramework || PlatformDetection.IsArmOrArm64Process || PlatformDetection.IsAlpine ? 14 : 15;
```
Comment (stephentoub) [inline src/Microsoft.VisualBasic.Core/tests/FinancialTests.cs]: Nit: s_precision

**Comment by cston** [comment]
@stephentoub, please take a look at the latest commit when you have a chance. The precision is now set for each test case. Thanks.

File: src/Microsoft.VisualBasic.Core/tests/FinancialTests.cs [hunk 3]
```diff
@@ -10,35 +10,27 @@ namespace Microsoft.VisualBasic.Tests
     [ActiveIssue(39888, TargetFrameworkMonikers.NetFramework)]
     public class FinancialTests
     {
-        private static bool IsNotArmOrAlpine() => !PlatformDetection.IsArmOrArm64Process && !PlatformDetection.IsAlpine;
+        // The accuracy of some numeric parsing and formatting depends on the platform.
```
Comment (stephentoub) [inline src/Microsoft.VisualBasic.Core/tests/FinancialTests.cs]: Nit: "the accuracy of some" => "the accuracy to which we can validate some" ?

File: src/Microsoft.VisualBasic.Core/tests/FinancialTests.cs [hunk 4]
```diff
@@ -103,20 +93,20 @@ public void IPmt_Invalid(double Rate, double Per, double NPer, double PV, double
         }
 
         [Theory]
-        [InlineData(new double[] { -1, 1 }, 0, 0, 0)]
-        [InlineData(new double[] { -70000.0, 22000.0, 25000.0, 28000.0, 31000.0 }, 0.1, 0.177435884421108, 0.177435884421108)]
-        [InlineData(new double[] { -10000.0, 6000.0, -2000.0, 7000.0, 1000.0 }, 0.1, 0.0866720474291716, 0.08667204742917164)]
-        [InlineData(new double[] { -30000.0, -10000.0, 25000.0, 12000.0, 15000.0 }, 0.1, 0.10928101434576, 0.10928101434575987)]
-        public void IRR(double[] ValueArray, double Guess, double expectedOld, double expectedNew)
+        [InlineData(new double[] { -1, 1 }, 0, 0, -0)]
+        [InlineData(new double[] { -70000.0, 22000.0, 25000.0, 28000.0, 31000.0 }, 0.1, 0.177435884421108, -0)]
```
Comment (stephentoub) [inline src/Microsoft.VisualBasic.Core/tests/FinancialTests.cs]: -0 ?

File: src/Microsoft.VisualBasic.Core/tests/FinancialTests.cs [hunk 5]
```diff
@@ -10,35 +10,27 @@ namespace Microsoft.VisualBasic.Tests
     [ActiveIssue(39888, TargetFrameworkMonikers.NetFramework)]
```
Comment (stephentoub) [inline src/Microsoft.VisualBasic.Core/tests/FinancialTests.cs]: Can this also be removed now?

**Pull request merged**